File: backends/nest/src/capsule-suggestion/capsule-suggestion.controller.ts
import { Body, Controller, HttpCode, Post } from '@nestjs/common';
import { CapsuleSuggestionService } from './capsule-suggestion.service';
import { CapsuleSuggestionRequestDto } from './dto/capsule-suggestion-request.dto';

@Controller('api/v1/capsule-suggestion')
export class CapsuleSuggestionController {
  constructor(private readonly service: CapsuleSuggestionService) {}

  @Post()
  @HttpCode(200)
  suggest(@Body() dto: CapsuleSuggestionRequestDto) {
    return this.service.suggest(dto.title);
  }
}

File: fullstacks/spring-mvc/src/main/java/com/hellotimepro/springmvc/web/CapsuleSuggestionController.java
package com.hellotimepro.springmvc.web;

import com.hellotimepro.springmvc.dto.Dtos.CapsuleSuggestion;
import com.hellotimepro.springmvc.dto.Dtos.CapsuleSuggestionRequest;
import com.hellotimepro.springmvc.dto.Dtos.Envelope;
import com.hellotimepro.springmvc.service.CapsuleSuggestionService;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v1")
public class CapsuleSuggestionController {
  private final CapsuleSuggestionService service;

  public CapsuleSuggestionController(CapsuleSuggestionService service) {
    this.service = service;
  }

  @PostMapping("/capsule-suggestion")
  public Envelope<CapsuleSuggestion> suggest(@Valid @RequestBody CapsuleSuggestionRequest req) {
    return Envelope.ok(service.suggest(req));
  }
}
